Core/Core/Inbox/MessageDetails/MessageDetailsAssembly.swift

```diff
@@ -20,10 +20,18 @@ import Foundation
 
 public enum MessageDetailsAssembly {
 
-    public static func makeViewController(env: AppEnvironment,
-                                          conversationID: String) -> UIViewController {
+    public static func makeViewController(
+        env: AppEnvironment,
+        conversationID: String,
+        allowArchive: Bool
```

Comment:
We should check if we can decide the value of this property inside the assembly or in the view model / interactor. The reason is that messages are frequently opened from push notifications where this url parameter won't be available.

Reply:
We don't have that information, hence this solution.